catalyst/core/callbacks/checkpoint.py

```diff
@@ -1,61 +1,83 @@
-from typing import Dict  # isort:skip
+from typing import Dict, Union  # isort:skip
 from collections import OrderedDict
 import os
 from pathlib import Path
 
-import safitty
-
 from catalyst import utils
-from catalyst.core import _State, Callback, CallbackOrder
+from catalyst.core import _State, Callback, CallbackNode, CallbackOrder
+
+
+def _load_checkpoint(*, filename, state: _State):
+    if os.path.isfile(filename):
+        print(f"=> loading checkpoint {filename}")
+        checkpoint = utils.load_checkpoint(filename)
+
+        if not state.stage_name.startswith("infer"):
+            state.epoch = checkpoint["epoch"] - 1
```

Comment:
it breaks backward compatibility (load older checkpoints), why not start epochs and iterations from 1 instead of 0?

Reply:
fixed